themes/default/content/blog/changes-to-import/index.md

```diff
@@ -0,0 +1,235 @@
+
+#### Code Comparison
+
+This code would have been generated by the older implementation:
+
+```typescript
+const my_bucket = new aws.s3.Bucket("my-bucket", {
+    acl: "private",
+    bucket: "my-bucket-3f85c54",
+    forceDestroy: false,
+}, {
+    protect: true,
+});
+```
+
+This code will be generated by the newer implementation:
+
+```typescript
+const my_bucket = new aws.s3.Bucket("my-bucket", {
+    arn: "arn:aws:s3:::my-bucket-3f85c54",
+    bucket: "my-bucket-3f85c54",
+    hostedZoneId: "Z3BJ6K6RIION7M",
+    requestPayer: "BucketOwner",
+}, {
+    protect: true,
+});
+```
+
+The above accurately reflects what the providers `Read` function has told us of the inputs set for this bucket, although due to the way the AWS provider happens to work we could elide all these properties. This is because the provider will use outputs saved in the state file and the new inputs to calculate the update diff, and a property that exists in the old output and is missing from the new inputs is simply left at the old output value. As none of the properties for `Bucket` are marked required the following also works and is equivalent:
+
+```
+const my_bucket = new aws.s3.Bucket("my-bucket", { }, { protect: true });
+```
+
+### Example: AWS EC2 Instances
+
+Previously, when trying to import an EC2 instance with `pulumi import`, you'd be presented with a list of required properties that weren't satisfied. The challenge here is that none of these fields are required by the schema, but at least one is required for the resource to be valid. Like above, the new behavior will read from the provider and inherit the missing properties, providing everything required to codegen. This provides a much cleaner and intuitive experience for the user.
+
+#### Old Behavior
+
+```shell
+aws:ec2:Instance (test):
+  error: aws:ec2/instance:Instance resource 'test' has a problem: Missing required argument: "instance_type": one of `instance_type,launch_template` must be specified. Examine values at 'Instance.InstanceType'.
+  error: aws:ec2/instance:Instance resource 'test' has a problem: Missing required argument: "launch_template": one of `ami,instance_type,launch_template` must be specified. Examine values at 'Instance.LaunchTemplate'.
+  error: aws:ec2/instance:Instance resource 'test' has a problem: Missing required argument: "ami": one of `ami,launch_template` must be specified. Examine values at 'Instance.Ami'.
+```
+
+#### New Behavior
+
+The import will be successful and we'll be presented with a rich resource in your language of choice.
+
+```typescript
+const test = new aws.ec2.Instance("test", {
+    ami: "ami-082b5a644766e0e6f",
+    associatePublicIpAddress: true,
+    availabilityZone: "us-west-2b",
+    capacityReservationSpecification: {
+        capacityReservationPreference: "open",
+    },
+    cpuCoreCount: 1,
+    cpuThreadsPerCore: 1,
+    creditSpecification: {
+        cpuCredits: "standard",
+    },
+    iamInstanceProfile: "",
```

Comment:
Aren't we eliding empty strings now?

Reply:
We should be, I'll double check this one.

Reply:
I think its getting confused because this is either a string or an object. We should fix this.

Reply:
https://github.com/pulumi/pulumi-terraform-bridge/issues/481 So we don't forget